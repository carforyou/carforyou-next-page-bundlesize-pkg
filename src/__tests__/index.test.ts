import fs from "fs"

import run from "../index"

describe("cli", () => {
  it("produces config with correct files and sizes", () => {
    const mockExit = jest.spyOn(process, "exit").mockImplementation()
    run([
      "jest",
      "./node_modules/.bin/jest",
      "--maxSize",
      "1 kB",
      "--buildDir",
      "./src/__tests__/.next",
    ])

    const config = fs
      .readFileSync("./src/__tests__/.next/next-page-bundlesize.config.json")
      .toString()
    expect(config).toMatchSnapshot()
    expect(mockExit).toHaveBeenCalledWith(0)
  })

  it("produces concatenated bundle file", () => {
    const mockExit = jest.spyOn(process, "exit").mockImplementation()
    run([
      "jest",
      "./node_modules/.bin/jest",
      "--maxSize",
      "1 kB",
      "--buildDir",
      "./src/__tests__/.next",
    ])

    const page = fs
      .readFileSync("./src/__tests__/.next/.bundlesize_")
      .toString()
    expect(page).toMatchSnapshot()
    expect(mockExit).toHaveBeenCalledWith(0)
  })

  it("fails when bundles are larger than the limit", () => {
    const mockExit = jest.spyOn(process, "exit").mockImplementation()
    run([
      "jest",
      "./node_modules/.bin/jest",
      "--maxSize",
      "1 b",
      "--buildDir",
      "./src/__tests__/.next",
    ])
    expect(mockExit).toHaveBeenCalledWith(1)
  })
})
