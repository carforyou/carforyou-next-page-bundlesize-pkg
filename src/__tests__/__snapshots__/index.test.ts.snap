// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli produces concatenated bundle file 1`] = `
"// _app.js
// index.js
"
`;

exports[`cli produces config with correct files and sizes 1`] = `"{\\"files\\":[{\\"path\\":\\"src/__tests__/.next/.bundlesize_\\",\\"maxSize\\":\\"1 kB\\"},{\\"path\\":\\"src/__tests__/.next/.bundlesize__app\\",\\"maxSize\\":\\"1 kB\\"}]}"`;
